### PR TITLE
Refactor interfaces

### PR DESCRIPTION
**Pull request by Barthelemy**

### CODE REVIEW

**Review by iravasen — APPROVED** [review]

File: Framework/include/QualityControl/CommonInterface.h [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+#include "QualityControl/QcInfoLogger.h"
+
+namespace o2::quality_control::core
+{
+
+/// \brief  Common interface for Check and Task Interfaces.
+///
+/// \author Barthelemy von Haller
+class CommonInterface
```
Comment (knopers8) [inline Framework/include/QualityControl/CommonInterface.h:32]: I am not sure if I like the name, it does not tell us what is it *common* for. How about `UserAlgorithmInterface` or `UserCodeInterface`?

File: Framework/include/QualityControl/CommonInterface.h [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+  /// \brief Configure the aggregator based on its name.
+  ///
+  /// The configuration of the object can't be done in the constructor because
+  /// ROOT needs an argument-less constructor when streaming it. Users can use
+  /// this method to configure their object. The name might be used to ask the configuration system
+  /// for specific parameters.
+  /// It is called each time mCustomParameters is updated.
+  ///
+  /// \param name The name of the task/aggregator/check.
```
Comment (knopers8) [inline Framework/include/QualityControl/CommonInterface.h:50]: This doc seems outdated

File: Modules/MFT/include/MFT/QcMFTReadoutCheck.h [hunk 3]
```diff
@@ -1,69 +1,69 @@
...
+  void resetVector(std::vector<int>& vector);
+  Quality checkQualityStatus(TH1F* histo, std::vector<int>& vector);
+  void writeMessages(TH1F* histo, std::vector<int>& vector, Quality checkResult);
+
+  ClassDefOverride(QcMFTReadoutCheck, 2);
+};
+
+} // namespace o2::quality_control_modules::mft
+
+#endif // QC_MFT_READOUT_CHECK_H
```
Comment (knopers8) [inline Modules/MFT/include/MFT/QcMFTReadoutCheck.h:69]: what happened here? end of line character fix?

**Review by knopers8 — CHANGES_REQUESTED** [review]
Thanks, i have only minor comments, it is good otherwise.

File: Modules/MFT/include/MFT/QcMFTReadoutCheck.h (hunk 3 above)
Comment (tomas-herman) [reply Modules/MFT/include/MFT/QcMFTReadoutCheck.h:69]: Also I see that the line 42 was not removed while in the other files it was deleted.

**Review by tomas-herman — APPROVED** [review]

**Review by knopers8 — APPROVED** [review]

File: Framework/CMakeLists.txt [hunk 4]
```diff
@@ -240,6 +242,7 @@ set(TEST_SRCS
     test/testRepoPathUtils.cxx
     test/testPolicyManager.cxx
     test/testQualitiesToTRFCollectionConverter.cxx
+    test/testCommonInterface.cxx
```
Comment (knopers8) [inline Framework/CMakeLists.txt:245]: consider updating it also here

**Pull request merged**